libmscore/undo.cpp

```diff
@@ -2472,16 +2472,40 @@ void MoveTremolo::redo(EditData*)
       oldC1 = trem->chord1();
       oldC2 = trem->chord2();
 
-      // Move tremolo
+      // Move tremolo away from old chords
       trem->chord1()->setTremolo(nullptr);
       trem->chord2()->setTremolo(nullptr);
+
+      // Delete old tremolo on c1 and c2, if present
+      if (c1->tremolo() && (c1->tremolo() != trem)) {
+            if (c2->tremolo() == c1->tremolo())
+                  c2->tremolo()->setChords(c1,c2);
+            else
+                  c1->tremolo()->setChords(c1,nullptr);
+            Tremolo* oldTremolo  = c1->tremolo();
+            c1->setTremolo(nullptr);
+            delete oldTremolo;
```

Comment:
It may be better not to delete this tremolo but remove it somehow in an undoable way. This tremolo may have appeared already in undo stack, for example because of some properties changes.

Reply:
I don't think that could happen.
I had worked on this part of the code in an effort to reduce memory leakages and then I found this issue: the issue itself is solved by the change at lines 2503-2508, but I included the additional code changes to prevent memory leaks.
These tremolos are created when a copy/clone is made as multiple copies, and they are basically discarded. Only the initial and the final tremolos are kept, while all the "intermediate" copies are leaked: if you look at the code before this PR, every reference to these tremolos is lost at the end of the function.
Some time has passed so I do not remember the fine details anymore, but you can check by adding a couple of line to output the pointers to the tremolos when they are created and deleted and then verify with the debugger which are actually the tremolos really appearing in the score, after the layout process. These "temporary" tremolos never appear in the score.
But to be honest, I do not know if there could be some strange corner cases in which these lines are not behaving properly.

Reply:
Well, then let's better have it in 3.4 to allow more testing in beta.

Reply:
Ok. :+1: